t/05-encrypt.t

```diff
@@ -0,0 +1,148 @@
+# vim:set ft= ts=4 sw=4 et:
+
+use Test::Nginx::Socket::Lua;
+use Cwd qw(cwd);
+
+repeat_each(2);
+
+plan tests => repeat_each() * blocks() * 5;
+
+my $pwd = cwd();
+
+our $MainConfig = qq{
+    lmdb_environment_path /tmp/test5.mdb;
+    lmdb_map_size 5m;
+    lmdb_encryption_key_file "12345678900987654321123456789002";
+};
+
+our $HttpConfig = qq{
+    lua_package_path "$pwd/lib/?.lua;;";
+};
+
+no_long_string();
+#no_diff();
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: simple set() / get()
+--- http_config eval: $::HttpConfig
+--- main_config eval: $::MainConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local l = require("resty.lmdb")
+
+            ngx.say(l.set("test", "value"))
+            ngx.say(l.get("test"))
+            ngx.say(l.get("test_not_exist"))
+        }
+    }
+--- request
+GET /t
+--- response_body
+true
+value
+nil
+--- no_error_log
+[error]
+[warn]
+[crit]
+
+
+
+=== TEST 2: clear using set()
+--- http_config eval: $::HttpConfig
+--- main_config eval: $::MainConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local l = require("resty.lmdb")
+
+            ngx.say(l.set("test", "value"))
+            ngx.say(l.set("test", nil))
+            ngx.say(l.get("test"))
+        }
+    }
+--- request
+GET /t
+--- response_body
+true
+true
+nil
+--- no_error_log
+[error]
+[warn]
+[crit]
+
+
+
+=== TEST 3: db_drop()
+--- http_config eval: $::HttpConfig
+--- main_config eval: $::MainConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local l = require("resty.lmdb")
+
+            ngx.say(l.set("test", "value"))
+            ngx.say(l.db_drop())
+            ngx.say(l.get("test"))
+        }
+    }
+--- request
+GET /t
+--- response_body
+true
+true
+nil
+--- no_error_log
+[error]
+[warn]
+[crit]
+
+
+
+=== TEST 4: db_drop(delete = true)
+--- http_config eval: $::HttpConfig
+--- main_config eval: $::MainConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local l = require("resty.lmdb")
+
+            ngx.say(l.set("test", "value"))
+            ngx.say(l.db_drop(true))
+            ngx.say(l.db_drop(true))
+            ngx.say(l.get("test"))
+        }
+    }
+--- request
+GET /t
+--- response_body
+true
+true
+nilunable to open DB for access: MDB_NOTFOUND: No matching key/data pair found
+nilunable to open DB for access: MDB_NOTFOUND: No matching key/data pair found
+--- no_error_log
+[error]
+[warn]
+[crit]
+
+=== TEST 5: works fine when not enabled
+--- http_config eval: $::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            ngx.say("good")
+        }
+    }
+--- request
+GET /t
+--- response_body
+good
+--- no_error_log
+[error]
+[warn]
+[crit]
```

Comment:
let's add a new test, to set a key with value `"unencrypted"`  in the encrypted lmdb env, and read the content of `/tmp/test5.mdb/data.mdb`, and verify the plaintext `"encrypted"` is not found in the file; similarly, add a counterpart test in 01-sanity.t, to verify with an unencrypted lmdb env, you can find the plaintext in `/tmp/test.mdb/data.mdb`.